### PR TITLE
Added missing "default_forum_layout" property for DiscordChannel

### PR DESCRIPTION
**Pull request by LaPepega**
# Description

Added missing "default_forum_layout" property for DiscordChannel
Fixes [#436](https://github.com/Aiko-IT-Systems/DisCatSharp/issues/436)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

**Comment by Lulalaby** [comment]
Thank you! Could you also check if it's in the modify / create function?
I don't remember it right now.

**Comment by TheNickOfMax** [comment]
@Lulalaby If you mean the ModifyChannelAsync and CreateChannelAsync methods, I believe it's not, but I can try adding it. (Sorry, I might do/understand something wrong, its my first PR)

**Comment by Lulalaby** [comment]
Yeah exactly!
If you want, try it, and if u need help, ask.
I'm a bit busy the next hours but after that I can help out, or you ask @TheXorog

**Comment by TheNickOfMax** [comment]
Uh, so I've been trying to wrap my head around the whole structure, and these methods use modify/create payload types to build queries.
I'm not sure if it's ok to edit them? Should I add `ForumLayout? defaultForumLayout` argument to these functions *and* a `defaultForumLayout` property for payload classes? And then there is the same thing for create and edit Models...
Not to be annoying, I just really don't want to break anything :)

**Comment by Lulalaby** [comment]
yeah that's exactly how to do it

**Comment by Lulalaby** [comment]
Take a look at https://github.com/Aiko-IT-Systems/DisCatSharp/commit/93aee4efc28aefdb6b1cda0beb42fdb0b8a4ef72

**Comment by LaPepega** [comment]
Great! Then I'll do it tomorrow.

**Comment by Lulalaby** [comment]
Sounds good

**Comment by LaPepega** [comment]
I noticed right now, wouldn't, `CreateGuildForumChannelAsync` be a better name for `CreateForumChannelAsync` method in [`DiscordApiClient`](https://github.com/Aiko-IT-Systems/DisCatSharp/blob/main/DisCatSharp/Net/Rest/DiscordApiClient.cs), since it uses guild_id? Analogous to `CreateGuildChannelAsync`

**Comment by Lulalaby** [comment]
If you look at https://github.com/Aiko-IT-Systems/DisCatSharp/pull/466/files, you'll see the missing places.
Also `CreateGuildForumChannelAsync` would be wrong because the method exists only on `DiscordGuild`, which implies it already.

**Comment by LaPepega** [comment]
That seems to be it.

**Comment by LaPepega** [comment]
> the method exists only on `DiscordGuild`, which implies it already.

I might be wrong, but i meant [this one](https://github.com/Aiko-IT-Systems/DisCatSharp/blob/f64429c9668f02a362f3681877597bad78866097/DisCatSharp/Net/Rest/DiscordApiClient.cs#L2366), which is in `DiscordApiClient`

**Comment by Lulalaby** [comment]
Ohhhhh. Yeah sure go for it!

**Comment by LaPepega** [comment]
Done! :)

**Comment by TheXorog** [comment]
lgtm

not quite familiar with how the modifying works for the most part though, so i trust your word that everythings implemented thats needed. what is implemented looks to be in line with everything else.

**Review by Lulalaby — CHANGES_REQUESTED** [review]
One small thing tho, audit log reason should always be the last arg

**Comment by Lulalaby** [comment]
@LaPepega can you work on the last feedback please

**Comment by LaPepega** [comment]
Sorry, i totally forgot, should be all good now though

**Comment by Lulalaby** [comment]
@TheXorog could u test?

File: DisCatSharp/Net/Rest/DiscordApiClient.cs [hunk 1]
```diff
@@ -2361,9 +2361,10 @@ internal async Task<DiscordChannel> CreateGuildChannelAsync(ulong guildId, strin
 	/// <param name="postCreateUserRateLimit">The per user post create rate limit.</param>
 	/// <param name="defaultAutoArchiveDuration">The default auto archive duration.</param>
 	/// <param name="defaultSortOrder">The default sort order.</param>
+	/// <param name="defaultLayout">The default layout.</param>
 	/// <param name="flags">The channel flags.</param>
 	/// <param name="reason">The reason.</param>
-	internal async Task<DiscordChannel> CreateForumChannelAsync(
+	internal async Task<DiscordChannel> CreateGuildForumChannelAsync(
```
Comment (TheXorog) [inline DisCatSharp/Net/Rest/DiscordApiClient.cs:2367]: I'm unsure why this method name would be changed, this is a breaking change.
Without it still is, but affects less people.

I don't know if this is a new change or whether i missed it last time. I apologize if it's the latter.

**Review by TheXorog — CHANGES_REQUESTED** [review]

**Comment by TheXorog** [comment]
> @TheXorog could u test?

Will do in about half an hour 👍

**Comment by TheXorog** [comment]
Tested with this:

```cs
[SlashCommand("forum", "Test.")]
public async Task DefaultForum(InteractionContext ctx, [Option("channel", "channel"), ChannelTypes(ChannelType.Forum)]DiscordChannel channel, [Option("layout", "layout")]ForumLayout layout)
{
    await channel.ModifyForumAsync(x => x.DefaultForumLayout = layout);
    await ctx.CreateResponseAsync(InteractionResponseType.DeferredChannelMessageWithSource, new DiscordInteractionResponseBuilder().AsEphemeral());
}

[SlashCommand("forum2", "Test.")]
public async Task DefaultForum2(InteractionContext ctx, [Option("layout", "layout")]ForumLayout layout)
{
    await ctx.Guild.CreateForumChannelAsync("test", ctx.Guild.GetChannel(1019898956316151850), defaultLayout: layout);
    await ctx.CreateResponseAsync(InteractionResponseType.DeferredChannelMessageWithSource, new DiscordInteractionResponseBuilder().WithContent("Success").AsEphemeral());
}
```

Seems all to be working.
Once i get some clarification on my review, this can be merged.

File: DisCatSharp/Net/Rest/DiscordApiClient.cs (hunk 1 above)
Comment (Lulalaby) [reply DisCatSharp/Net/Rest/DiscordApiClient.cs:2367]: I approved it because it's an internal change not facing to devs ever

**Review by Lulalaby — APPROVED** [review]

**Comment by TheXorog** [comment]
Thanks for your contribution! @LaPepega

**Pull request merged**